pkg/flowkit/services/project.go

```diff
@@ -200,9 +208,29 @@ func (p *Project) Deploy(network string, update bool) ([]*contracts.Contract, er
 			deployErr = true
 			continue
 		} else if exists {
-			tx, err = flowkit.NewUpdateAccountContractTransaction(targetAccount, contract.Name(), contract.TranspiledCode())
-			if err != nil {
-				return nil, err
+			//only update contract if there is diff
+			if noDiffInContract {
+				p.logger.Info(fmt.Sprintf(
+					"no diff found in %s, skipping update",
+					contract.Name(),
+				))
+				continue
+			} else {
+				if !force {
+					edits := myers.ComputeEdits(span.URIFromPath(contract.Source()), string(existingContract), contract.Code())
+					diff := fmt.Sprint(gotextdiff.ToUnified("deployed", "changes", string(existingContract), edits))
+
+					fmt.Println("DIFF of " + contract.Name())
+					fmt.Println(diff)
+					fmt.Println("------------------------")
+					fmt.Println(" press <ENTER> to update")
+					fmt.Scanln()
```

Comment:
You shouldn't use `fmt.Println` anywhere in the service layer, if you need to log something you should use a logger, if you need an interactive prompt you should use the prompt in the `output` package https://github.com/onflow/flow-cli/blob/master/pkg/flowkit/output/prompt.go.

Reply:
don't need this anymore since we are not doing prompts